File: src/main/java/me/zax71/lightHub/listeners/PlayerLogin.java
package me.zax71.lightHub.listeners;

import net.kyori.adventure.text.Component;
import net.kyori.adventure.text.format.NamedTextColor;
import net.kyori.adventure.text.serializer.ansi.ANSIComponentSerializer;
import net.minestom.server.MinecraftServer;
import net.minestom.server.adventure.audience.Audiences;
import net.minestom.server.entity.GameMode;
import net.minestom.server.entity.Player;
import net.minestom.server.event.EventListener;
import net.minestom.server.event.player.PlayerLoginEvent;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

import static me.zax71.lightHub.Main.config;
import static me.zax71.lightHub.Main.configUtils;
import static me.zax71.lightHub.Main.hubInstance;
import static me.zax71.lightHub.Main.logger;


public class PlayerLogin implements EventListener<PlayerLoginEvent> {
    @Override
    public @NotNull Class<PlayerLoginEvent> eventType() {
        return PlayerLoginEvent.class;
    }

    @Override
    public @NotNull Result run(@NotNull PlayerLoginEvent event) {
        Player player = event.getPlayer();
        event.setSpawningInstance(hubInstance);
        player.setRespawnPoint(Objects.requireNonNull(configUtils.getPosFromConfig(config.node("spawnPoint"))));
        player.setGameMode(GameMode.ADVENTURE);

        // Tell players, and the log, that someone joined
        Component playerJoinMessage = player.getName().append(Component.text(" joined the server")).color(NamedTextColor.YELLOW);
        Audiences.players().sendMessage(playerJoinMessage);
        logger.info(ANSIComponentSerializer.ansi().serialize(playerJoinMessage));

        // Set team
        player.setTeam(MinecraftServer.getTeamManager().getTeam("noCollision"));
        return Result.SUCCESS;
    }
}

File: src/main/kotlin/me/zax71/lightHub/listeners/PlayerLogin.kt
package me.zax71.lightHub.listeners

import me.zax71.lightHub.instances.DefaultInstance
import me.zax71.lightHub.inventories.GameMenu
import net.kyori.adventure.text.Component
import net.kyori.adventure.text.format.TextDecoration
import net.minestom.server.coordinate.Pos
import net.minestom.server.entity.GameMode
import net.minestom.server.event.EventListener
import net.minestom.server.event.player.PlayerLoginEvent
import net.minestom.server.item.ItemStack
import net.minestom.server.item.Material

class PlayerLogin : EventListener<PlayerLoginEvent> {
    /*
    Use to pass variables in


    private final Instance lobby;
    public PlayerLogin(Instance lobby){
        this.playerManager = manager;
        this.lobby = lobby;
    }
     */
    override fun eventType(): Class<PlayerLoginEvent> {
        return PlayerLoginEvent::class.java
    }

    override fun run(event: PlayerLoginEvent): EventListener.Result {
        // When event is run
        val player = event.player

        // Spawning
        event.setSpawningInstance(DefaultInstance.INSTANCE!!)
        player.respawnPoint = Pos(0.0, -60.0, 0.0)
        player.gameMode = GameMode.CREATIVE

        // Adding compass to inventory
        val inventory = player.inventory
        val compass = ItemStack.builder(Material.COMPASS)
            .displayName(Component.text("Navigator").decoration(TextDecoration.ITALIC, false)).build()
        inventory.setItemStack(4, compass)

        player.inventory.addInventoryCondition { playerInventory, slot, _, inventoryConditionResult ->
            // Moving items is bad
            inventoryConditionResult.isCancel = true

            if (slot == 4) {
                playerInventory.openInventory(GameMenu.inventory!!)
            }

        }
        return EventListener.Result.SUCCESS
    }
}